src/Soil.jl

```diff
@@ -154,8 +135,47 @@ function make_update_aux(model::RichardsModel)
             effective_saturation(ν, Y.soil.ϑ_l, θ_r),
         )
         @. p.soil.ψ = pressure_head(vg_α, vg_n, vg_m, θ_r, Y.soil.ϑ_l, ν, S_s)
-        return update_aux!
     end
+    return update_aux!
+end
+
+##### Types and methods for handling boundary conditions and source terms
+abstract type AbstractSoilBoundaryConditions{FT <: AbstractFloat} end
+
+abstract type AbstractSoilSource{FT <: AbstractFloat} end
+
+"""
+    SoilConfiguration{FT, BCT, ST} <: AbstractConfiguration{FT}
+
+The Configuration object for the soil model, holding boundary conditions
+and source terms, which are used for dispatching on within the soil RHS.
+"""
+struct SoilConfiguration{FT, BCT<: AbstractSoilBoundaryConditions{FT},ST} <: AbstractConfiguration{FT}
```

Comment:
As we discussed, I think it doesn't seem like individual models need an internal `Configuration` struct that then stores the source and bcs. We can directly store these in the model instead